Assets/Tests/InputSystem/CoreTests_Actions.cs

```diff
@@ -1616,6 +1616,289 @@ public void Actions_CanQueryIfPerformedInCurrentFrame()
+            Assert.That(defaultAction.WasPressedThisFrame(), Is.False);
+            Assert.That(defaultAction.WasReleasedThisFrame(), Is.False);
+            Assert.That(defaultAction.WasPerformedThisFrame(), Is.False);
+            Assert.That(defaultAction.WasUnperformedThisFrame(), Is.False);
+
+            Assert.That(pressTrace, Is.Empty);
+            Assert.That(pressAction.WasPressedThisFrame(), Is.False);
+            Assert.That(pressAction.WasReleasedThisFrame(), Is.False);
+            Assert.That(pressAction.WasPerformedThisFrame(), Is.False);
+            Assert.That(pressAction.WasUnperformedThisFrame(), Is.False);
+
+            Assert.That(holdTrace, Performed(holdAction));
+            Assert.That(holdAction.WasPressedThisFrame(), Is.False);
+            Assert.That(holdAction.WasReleasedThisFrame(), Is.False);
+            Assert.That(holdAction.WasPerformedThisFrame(), Is.True);
+            Assert.That(holdAction.WasUnperformedThisFrame(), Is.False);
+
+            defaultTrace.Clear();
+            pressTrace.Clear();
+            holdTrace.Clear();
+
+            // Keep holding button.
+            InputSystem.Update();
+
+            Assert.That(defaultTrace, Is.Empty);
+            Assert.That(defaultAction.WasPressedThisFrame(), Is.False);
+            Assert.That(defaultAction.WasReleasedThisFrame(), Is.False);
+            Assert.That(defaultAction.WasPerformedThisFrame(), Is.False);
+            Assert.That(defaultAction.WasUnperformedThisFrame(), Is.False);
+
+            Assert.That(pressTrace, Is.Empty);
+            Assert.That(pressAction.WasPressedThisFrame(), Is.False);
+            Assert.That(pressAction.WasReleasedThisFrame(), Is.False);
+            Assert.That(pressAction.WasPerformedThisFrame(), Is.False);
+            Assert.That(pressAction.WasUnperformedThisFrame(), Is.False);
+
+            Assert.That(holdTrace, Is.Empty);
+            Assert.That(holdAction.WasPressedThisFrame(), Is.False);
+            Assert.That(holdAction.WasReleasedThisFrame(), Is.False);
+            Assert.That(holdAction.WasPerformedThisFrame(), Is.False);
+            Assert.That(holdAction.WasUnperformedThisFrame(), Is.False);
+
+            defaultTrace.Clear();
+            pressTrace.Clear();
+            holdTrace.Clear();
+
+            // Release button, the actions should cancel. The hold action should now be considered unperformed
+            // since it was held long enough to be performed.
+            Release(gamepad.buttonSouth);
+
+            Assert.That(defaultTrace, actionType != InputActionType.PassThrough
+                ? Canceled(defaultAction)
+                : Performed(defaultAction));
+            Assert.That(defaultAction.WasPressedThisFrame(), Is.False);
+            Assert.That(defaultAction.WasReleasedThisFrame(), Is.True);
+            Assert.That(defaultAction.WasPerformedThisFrame(), Is.EqualTo(actionType == InputActionType.PassThrough));
```

Comment:
Unperformed isn't that useful for Pass-Through actions since those actions don't go through any kind of button-like phase progression. It's just always Performed, even when the control goes back to default.